scripts/ci/run-jenkins.sh

```diff
@@ -34,7 +34,17 @@ if [[ -n "${ghprbPullId}" ]] && [[ ${label} == w* ]];
     # we don't run the test suite on Windows PRs, we just ensure the build succeeds, so end here
 fi
 ${TESTCMD} --label=mini --timeout=5m make -w -C mono/mini -k check
+if [[ ${label} == w* ]]; then
+	# Create a temporary config which maps libtest to the MSVC built libtest.dll. Without this the P/Invoke tests in mono/tests/ won't find libtest.dll.
+	MONO_CONFIG=$(mktemp --suffix=.xml)
+	echo "<configuration><dllmap dll=\"libtest\" target=\"${WORKSPACE}\\msvc\\${PLATFORM}\\bin\\Release\\libtest.dll\"/></configuration>" > "$MONO_CONFIG"
+	export MONO_CONFIG=$(cygpath -w -a "$MONO_CONFIG")
+fi
 ${TESTCMD} --label=runtime --timeout=120m make -w -C mono/tests -k test-wrench V=1 CI=1
+if [[ ${label} == w* ]] && [[ -n "$MONO_CONFIG" ]] && [[ -f "$MONO_CONFIG" ]]; then
+	rm -f "$MONO_CONFIG"
+	unset MONO_CONFIG
+fi
```

Comment:
I agree it would be nicer to have this done in the Makefile. How about making sure `tests-config` is always read for all tests (via `MONO_CONFIG`) and add 

```xml
<dllmap dll="libtest" target="@MONO_LIBTEST_PATH@" />
```

to `tests-config.in`? The Makefile would set `MONO_LIBTEST_PATH` to `./libtest.dylib` (or `.so`) unless `MONO_LIBTEST_PATH` has already been set from outside the Makefile. Then we can control whether the MSVC built DLL should be used by simply setting `MONO_LIBTEST_PATH` in `run-jenkins.sh`, just like already is done for the MSVC built `mono-sgen.exe`.

Let me know if this sounds ok and I will update the PR.